app/cloud_storage.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/python2.7
+# Copyright 2018 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base64
+import datetime
+import StringIO
+import time
+import urllib
+
+from apiclient.http import MediaIoBaseUpload
+from google.appengine.api import app_identity
+from googleapiclient.discovery import build
+from oauth2client.client import GoogleCredentials
+
+import config
+import utils
+
+
+# Lifetime of the generated signed URL.
+SIGNED_URL_LIFETIME = datetime.timedelta(minutes=10)
```

Comment:
> Could this be passed to sign_url()?

Done.

> Also, in low-bandwidth situations, I can imagine it being a slow download in cases with big files. Maybe make it 20 or 30 minutes instead of 10?

https://cloud.google.com/storage/docs/access-control/signed-urls says:
> The server rejects any requests received after this timestamp.

So I believe expiration is checked when the request is received. So it should be OK if it expires during the download.